src/AbstractFirestoreRepository.ts

```diff
@@ -108,14 +108,15 @@ export abstract class AbstractFirestoreRepository<T extends IEntity>
     if (!doc.exists) {
       return null;
     }
+    
     // tslint:disable-next-line:no-unnecessary-type-assertion
     const entity = plainToClass(
-      this.colMetadata.entity as any,
+      this.colMetadata.entity,
       this.transformFirestoreTypes(doc.data() as T)
     ) as T;
-    if (this.collectionType === FirestoreCollectionType.collection) {
```

Comment:
If this check is not needed anymore, shouldn't we remove the `collectionType` field altogether?